packages/gatsby/src/utils/flags.ts

```diff
@@ -0,0 +1,39 @@
+// Does this experiment run for only builds
+type executingCommand = "build" | "develop" | "all"
+
+export interface IFlag {
+  name: string
+  env: string
+  description: string
+  command: executingCommand
+  includedFlags?: Array<string>
+  umbrellaIssue?: string
+  noCi?: boolean
+}
+
+const activeFlags: Array<IFlag> = [
+  {
+    name: `FAST_DEV`,
+    env: `GATSBY_EXPERIMENTAL_FAST_DEV`,
+    command: `develop`,
+    description: `Enable all experiments aimed at improving develop server start time`,
+    includedFlags: [`DEV_SSR`, `QUERY_ON_DEMAND`],
+  },
+  {
+    name: `DEV_SSR`,
+    env: `GATSBY_EXPERIMENTAL_DEV_SSR`,
+    command: `develop`,
+    description: `SSR pages on full reloads during develop. Helps you detect SSR bugs and fix them without needing to do full builds.`,
+    umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/28138`,
+  },
+  {
+    name: `QUERY_ON_DEMAND`,
+    env: `GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND`,
+    command: `develop`,
+    description: `Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.`,
+    umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/27620`,
+    noCi: true,
+  },
+]
```

Comment:
Can we add `experimental` to the list? If any "experimental" flag is used, we can show a message to show that it might break their site.

When experiments are set to false, we're pretty confident it will work, but it might have some inconveniences.

```suggestion
const activeFlags: Array<IFlag> = [
  {
    name: `FAST_DEV`,
    env: `GATSBY_EXPERIMENTAL_FAST_DEV`,
    command: `develop`,
    description: `Enable all experiments aimed at improving develop server start time`,
    includedFlags: [`DEV_SSR`, `QUERY_ON_DEMAND`],
    experimental: false,
  },
  {
    name: `DEV_SSR`,
    env: `GATSBY_EXPERIMENTAL_DEV_SSR`,
    command: `develop`,
    description: `SSR pages on full reloads during develop. Helps you detect SSR bugs and fix them without needing to do full builds.`,
    umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/28138`,
    experimental: false,
  },
  {
    name: `QUERY_ON_DEMAND`,
    env: `GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND`,
    command: `develop`,
    description: `Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.`,
    umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/27620`,
    noCi: true,
    experimental: false,
  },
]
```

Reply:
yeah I like this — we need some heuristics around when to use this & when to disable it so it sends a consistent message e.g. experimental === very few people have tested it and/or there's serious known bugs — once bugs are mostly resolve & 50+ people have tested it, we remove it.